Assets/HoloToolkit/Utilities/Scripts/Solvers/SolverConstantViewSize.cs

```diff
@@ -0,0 +1,189 @@
+//
+using UnityEngine;
+using System.Collections;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    ///   ConstantViewSize solver scales to maintain a constant size relative to the view (currently tied to the Camera)
+    /// </summary>
+    public class SolverConstantViewSize : Solver
+    {
+        #region public members
+        [Tooltip("The object take up this percent vertically in our view (not technically a percent use 0.5 for 50%)")]
+        public float TargetViewPercentV = 0.5f;
+
+        [Tooltip("If the object is closer than MinDistance, the distance used is clamped here")]
+        public float MinDistance = 0.5f;
+
+        [Tooltip("If the object is farther than MaxDistance, the distance used is clamped here")]
+        public float MaxDistance = 3.5f;
+
+        [Tooltip("Minimum scale value possible (world space scale)")]
+        public float MinScale = 0.01f;
+
+        [Tooltip("Maximum scale value possible (world space scale)")]
+        public float MaxScale = 100f;
+
+        [Tooltip("Used for dead zone for scaling")]
+        public float ScaleBuffer = 0.01f;
+
+        [Tooltip("If you don't trust or don't like the auto size calculation, specify a manual size here. 0 is ignored")]
+        public float ManualObjectSize = 0;
+        public enum ScaleStateEnum
+        {
+            Static,
+            Shrinking,
+            Growing
+        }
+
+        [HideInInspector]
+        public ScaleStateEnum ScaleState = ScaleStateEnum.Static;
+
+        // 0 to 1 between MinScale and MaxScale.  If currently < max, then scaling is being applied.
+        // This value is subject to inaccuracies due to smoothing/interpolation/momentum
+        public float CurrentScalePercent
+        {
+            get
+            {
+                return objectScalePercent;
+            }
+        }
+
+        // 0 to 1 between MinDistance and MaxDistance.  If current < max, object is potentially on a surface [or some other condition like interpolating] (since it may still be on surface, but scale percent may be clamped at max)
+        // This value is subject to inaccuracies due to smoothing/interpolation/momentum
+        public float CurrentDistancePercent
+        {
+            get
+            {
+                return objectDistancePercent;
+            }
+        }
+        #endregion
+
+        #region private members
+        private float fovScalar = 1f;
+        private float objectSize = 1f;
+        private float objectScalePercent = 1f;
+        private float objectDistancePercent = 1f;
+        #endregion
+
+        protected override void Start()
+        {
+            base.Start();
+
+            float baseSize = CalculateObjectSize();
+
+            if (baseSize > 0)
+            {
+                objectSize = baseSize;
+            }
+            else
+            {
+                Debug.LogError("ConstantViewSize: Object base size calculate was 0, defaulting to 1");
```

Comment:
If we're just defaulting the value, is it still an error, or could this just be a warning?
An error would imply that the system wouldn't continue to function if we get to this point.